solidity/contracts/AugurConstantProductShareToken.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.8.29;
+
+import { IERC20 } from "./IERC20.sol";
+import { ERC1155 } from "./ERC1155.sol";
+import { Constants } from "./Constants.sol";
+import { TokenId } from "./TokenId.sol";
+import { IOICash } from "./IOICash.sol";
+
+contract AugurConstantProductShareToken is ERC1155 {
```

Comment:
Does this actually need to be an ERC1155?  I think the minimum requirement is that a user can "roll their keys", but I don't know if we need full 1155 support?  I'm not sure if it helps us to remove ERC-1155 dependency though.

Reply:
It probably doesn't need the full ERC1155 but I think we wouldn't gain much by ripping out functionality. There shouldn't be much overhead of those features in its operation within our system. Its a good mental construct too for understanding how the system is operating and what state looks like.